### ISSUE TITLE
Simplify 'change require/recommend and run test again' workflow

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#2821]
Let's say I do `tmt try` and I see that the test failed due `command not found`. I can easily fix that by adding missing dependency to 'require' or 'recommend'.

But to test again I need:
1. Press 't' which will discover test and dependencies again. Then it will FAIL again as nothing installed new dependencies
2. Press 'p' to run prepare step, which will install new dependencies (and will run all other prepare phases)
3. Press 't' to run the test again

Note that running 'prepare' again my fail as its phases aren't expected to be idempotent (in "real" tmt runs they are executed just once) or they just waste time (if they run too long).

I'd expect that 't' should be able to detect that require/recommend has changed and run just the necessary install phase without any additional user output. The flow will then be: 1. Press 't' to reflect changes and run test again.

**Comment by psss** [comment]
Yeah, makes sense to simplify this scenario, adding to the backlog.
